File: Challenges/Coding World Cup/2023/Day 6 - PAK vs SL - Best Workers.cpp
long long bestWorkers(int n, int k, vector<int> &a)
{
    // Write your code here.
    long long sum = 0;

    long long currSum = 0;
    for (int i = 0; i < k; i++)
    {
        currSum += a[i];
        sum = currSum;
    }
    for (int i = k; i < n; i++)
    {
        currSum -= a[i - k];
        currSum += a[i];
        sum = max(sum, currSum);
    }

    return sum;
}

File: Challenges/Coding World Cup/2023/Day 6 - PAK vs SL - Best Workers.java
public class Solution {
    static long bestWorkers(int n, int k, int[] a) {
        long maxProductivity = 0;
        long currentProductivity = 0;

        for (int i = 0; i < k; i++) {
            currentProductivity += a[i];
        }
        maxProductivity = currentProductivity;

        for (int i = k; i < n; i++) {
            currentProductivity += a[i] - a[i - k];
            maxProductivity = Math.max(maxProductivity, currentProductivity);
        }

        return maxProductivity;
    }
}
